lib/std/jsbigints.nim

```diff
@@ -0,0 +1,159 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func newJsBigInt*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert newJsBigInt(1234567890) == newJsBigInt"1234567890"
+
+func newJsBigInt*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert newJsBigInt"-1" == newJsBigInt"1" - newJsBigInt"2"
+
+func toCstring*(this: JsBigInt; radix: int): cstring {.importjs: "#.toString(#)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+  runnableExamples:
+    doAssert newJsBigInt"2147483647".toCstring(2) == "1111111111111111111111111111111".cstring
+
+func toCstring*(this: JsBigInt): cstring {.importjs: "#.toString()".} # asserted on $
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string =
+  ## Return a string representation of `JsBigInt`.
+  runnableExamples: doAssert $newJsBigInt"1024" == "1024"
+  $toCstring(this)
+
+func toInt*(bits: int; a: JsBigInt): JsBigInt {.importjs: "BigInt.asIntN(#, #)".} =
```

Comment:
since we're re-naming the API, might as well use the more useful/intuitive order of arguments that works better with UFCS/MCS; use this:
```nim
func toInt*(a: JsBigInt, bits: int): JsBigInt ...
```

also the name doesn't convey the meaning much, how about:
changing `toInt => wrapToInt` and `toUint => wrapToUint` ?

Reply:
@juancarlospaco what about this point?
I want to be able to write with UFCS / MCS, eg:
`getBigInt(). wrapToInt(32)` so I'd prefer flipping the order of arguments